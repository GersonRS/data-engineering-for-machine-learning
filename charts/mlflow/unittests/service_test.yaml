suite: test service

templates:
  - service.yaml

tests:
  - it: should use different port number when we change it
    set:
      service.port: 80
    asserts:
      - equal:
          path: spec.ports[0].port
          value: 80
      - equal:
          path: spec.ports[0].targetPort
          value: 80

  - it: should use different port name when we change it
    set:
      service.name: differentname
    asserts:
      - equal:
          path: spec.ports[0].name
          value: differentname

  - it: should match snapshot of default values
    set:
      service.annotations:
        test-key: test-value
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
