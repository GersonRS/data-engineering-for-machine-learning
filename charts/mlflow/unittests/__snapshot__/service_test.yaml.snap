should match snapshot of default values:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        test-key: test-value
      labels:
        app: mlflow
        app.kubernetes.io/instance: mlflow
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: mlflow
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: mlflow-1.0.0
        version: 1.0.0
      name: mlflow
    spec:
      ports:
      - name: http
        port: 5000
        protocol: TCP
        targetPort: 5000
      selector:
        app: mlflow
        app.kubernetes.io/instance: mlflow
        app.kubernetes.io/name: mlflow
      type: ClusterIP
